### PR TITLE
Bug - fix a few animations

### PR DESCRIPTION
**Pull request by jasonlundien**
This PR fixes animations that were NOT working: beatFade, flip, spinPulse, and spinReverse.

**Reviewer:**
@robmadole -- Will you review for accuracy of code?

**Dependencies:**
none, but was brought up in a Help Scout ticket

**Tests Added**
yes, also added tests for fixes.

### CODE REVIEW

**Comment by robmadole** [comment]
@jasonlundien do we need a test for `spinPulse` and `spinReverse`?

**Comment by robmadole** [comment]
Also, we probably need to update the [TypeScript .d.ts](https://github.com/FortAwesome/react-fontawesome/blob/bug-fix-a-few-animations/index.d.ts#L23-L52) file too with the changes.

**Comment by jasonlundien** [comment]
@robmadole -- I did NOT include those (`spinPulse` and `spinReverse` ) because the `beatFade` was of the same format.  When looking at the test file, `FontAwesomeIcon.test.js`, I noticed we were NOT testing each animation.  If you, you would like or prefer, I can create a test for each one including `spinPulse` and `spinReverse` ?

**Comment by robmadole** [comment]
@jasonlundien yeah let's add some tests for all of those. It might be overkill but we've missed a few of these types of things recently (and by "we" I mean "me" ;) so it would be good to add those please. Thank you, sir.

File: index.d.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ export interface FontAwesomeIconProps extends BackwardCompatibleOmit<SVGAttribut
   className?: string
   color?: string
   spin?: boolean
+  spinPulse?: boolean
```
Comment (jasonlundien) [inline index.d.ts:31]: `beatFade` and `flip` are already present.  `spinPulse` and `spinReverse` were the only animations missing.

File: index.d.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ export interface FontAwesomeIconProps extends BackwardCompatibleOmit<SVGAttribut
...
+  spinReverse?: boolean
```
Comment (jasonlundien) [inline index.d.ts:32]: `beatFade` and `flip` are already present.  `spinPulse` and `spinReverse` were the only animations missing.

**Review by robmadole — APPROVED** [review]

**Pull request merged**